crates/iota-types/src/move_package.rs

```diff
@@ -2,6 +2,37 @@
 // Modifications Copyright (c) 2024 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
+//! Move package
+//!
+//! This module contains the [MovePackage] and types necessary for describing
+//! its update behavior and linkage information for module resolution during
+//! execution.
+//!
+//! The code contains terminology that may be confusing for the uninitiated,
+//! like `Module ID`, `Package ID`, `Storage ID` and `Runtime ID`. For avoidance
+//! of doubt these concepts are defined as so:
+//! - `Package ID` is the [ObjectID] representing the address by which the given
+//!   package may be found in storage.
+//! - `Storage ID` represents the [ObjectID] by which an object may be found in
+//!   storage. Be that a [MovePackage], [CompiledModule] or just any shared
+//!   object.
```

Comment:
I guess also immutable and owned objects.
```suggestion
//!   storage. Be that a [MovePackage], [CompiledModule] or just any object.
```

Reply:
Fixed in: 859e18f20cc0c938824bd54b1fc5524b77a37aae
Moved them into ObjectId for even better docs.